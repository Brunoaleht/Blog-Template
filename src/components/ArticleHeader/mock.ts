import { ArticleHeaderProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  title: 'ReactHook Studies',
  slug: 'react-hook-studies',
  cover: {
    coverData: {
      attributes: {
        alternativeText: null,
        url: 'https://res.cloudinary.com/dxldesniw/image/upload/v1677769491/thumbnail_MKS_entrevista_87872be63c_60bb358adf.jpg',
      },
    },
  },
  excerpt:
    'I made this project using various react hooks, such as useEffect to make a call to the api, use State to keep my data in state, useContext and useReducer to do the work in the general context in the shopping cart.',
  author: {
    authorData: {
      id: '1',
      attributes: {
        displayName: 'Bruno Alexandre',
        slug: 'bruno-alexandre',
      },
    },
  },
  categories: {
    categoryData: [
      {
        id: '2',
        attributes: {
          displayName: 'ReactJs',
          slug: 'react-js',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'JavaScript',
          slug: 'java-script',
        },
      },
    ],
  },
} as ArticleHeaderProps;
