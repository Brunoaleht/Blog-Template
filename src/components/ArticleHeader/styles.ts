import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';
export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.large};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.fonts.sizes.medium};
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
