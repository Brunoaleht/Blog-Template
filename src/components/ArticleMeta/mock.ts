import { ArticleMetaProps } from '.';

export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    authorData: {
      id: '1',
      attributes: {
        displayName: 'Bruno Alexandre',
        slug: 'bruno-alexandre',
      },
    },
  },
  categories: {
    categoryData: [
      {
        id: '2',
        attributes: {
          displayName: 'ReactJs',
          slug: 'react-js',
        },
      },
      {
        id: '1',
        attributes: {
          displayName: 'JavaScript',
          slug: 'java-script',
        },
      },
    ],
  },
} as ArticleMetaProps;
