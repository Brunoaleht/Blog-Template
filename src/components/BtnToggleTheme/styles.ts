import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 2;
  `}
`;
export const Label = styled.label`
  ${() => css`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    overflow: hidden;
    font-size: 0;
    color: rgba(0, 0, 0, 0);
  `}
`;
export const BtnInput = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
    }
    &:focus + ${Slider} {
      border: solid 1px ${theme.colors.secondary};
    }
    &:checked + ${Slider}::before {
      transform: translateX(1.95rem);
      background: ${theme.colors.primary};
    }
  `}
`;
export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    cursor: pointer;
    background: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;
    border: solid 1px ${theme.colors.darkerGray};
    border-radius: 2rem;
    &::before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.1rem;
      background: ${theme.colors.textDark};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
