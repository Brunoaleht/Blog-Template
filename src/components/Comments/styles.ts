import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
