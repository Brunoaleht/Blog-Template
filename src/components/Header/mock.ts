import { HeaderProps } from '.';
export default {
  blogName: 'Bruno Alexandre',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Possimus doloremque nulla eius aperiam modi error facere, commodi.',
  logo: {
    logoData: {
      attributes: {
        alternativeText: null,
        url: 'https://res.cloudinary.com/dxldesniw/image/upload/v1677869914/logo1_6dca30fa4b.png',
      },
    },
  },
  showText: true,
} as HeaderProps;
