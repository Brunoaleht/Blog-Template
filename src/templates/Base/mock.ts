import { BaseProps } from '.';

export default {
  setting: {
    settingData: {
      identificador: '1',
      attributes: {
        blogName: 'Bruno Alexandre',
        blogDescription: 'Blog de ideias e de projetos realizados',
        textFooter: 'feito com amor',
        logo: {
          logoData: {
            attributes: {
              alternativeText: null,
              url: 'https://res.cloudinary.com/dxldesniw/image/upload/v1677869914/logo1_6dca30fa4b.png',
            },
          },
        },
        menuLink: [
          {
            id: '1',
            link: '#about',
            text: 'About',
            newTab: false,
          },
          {
            id: '3',
            link: '#skills',
            text: 'Skills',
            newTab: false,
          },
          {
            id: '2',
            link: '#contact',
            text: 'Contact',
            newTab: true,
          },
        ],
      },
    },
  },
  children: 'Ol√° Mundo',
} as BaseProps;
