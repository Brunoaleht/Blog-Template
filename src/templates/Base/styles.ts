import styled, { css } from 'styled-components';

export const Wrapper = styled.div``;
export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xhuge};
    padding: 0 ${theme.spacings.large};
  `}
`;
export const SearchContainer = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xlarge};
    padding: 0 ${theme.spacings.large};
    max-width: 120rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;
export const SearchInput = styled.input`
  ${({ theme }) => css`
    width: 50rem;
    padding: 0.5rem ${theme.spacings.small};
    @media ${theme.media.lteMedium} {
      width: 36rem;
    }
    @media ${theme.media.lteSmall} {
      width: 30rem;
    }
  `}
`;
export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xhuge};
  `}
`;

export const FooterContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding-top: ${theme.spacings.small};
  `}
`;
