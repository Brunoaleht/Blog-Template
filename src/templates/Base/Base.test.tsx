import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Base, BaseProps } from '.';
import mock from './mock';

const props: BaseProps = mock;

describe('<Base />', () => {
  it('should render Base with Elements: Header, Menu, Footer', () => {
    renderTheme(<Base {...props} />);

    expect(
      screen.getByRole('img', { name: 'Bruno Alexandre - Blog de ideias e de projetos realizados' }),
    ).toBeInTheDocument();
    expect(screen.getByLabelText('Open/Close')).toBeInTheDocument();
    expect(screen.getByText('feito com amor')).toBeInTheDocument();
    expect(screen.getByLabelText('Go to top')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<Base {...props} />);
    expect(container).toMatchSnapshot();
  });
});
