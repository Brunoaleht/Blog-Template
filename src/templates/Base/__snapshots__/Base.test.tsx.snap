// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should match snapshot 1`] = `
.c17 {
  position: fixed;
  background: #000000;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 3.2rem;
  padding-left: 3.2rem;
  padding-bottom: 5.6rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c11 .c5 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c12 p {
  margin-top: 0;
}

.c9 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  margin-bottom: 1.6rem;
  color: #ffffff;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c9:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c2 {
  background: #000000;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -30rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c4 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c8 {
  list-style: none;
  padding: 0;
}

.c0 {
  z-index: 2;
  position: fixed;
  top: 2rem;
  width: 4rem;
  height: 4rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 1rem;
  background: #000000;
  color: #ffffff;
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c0 > svg:hover {
  color: #dc143c;
}

.c15 {
  text-align: center;
  border-top: solid 0.1rem #dddddd;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
}

.c16 {
  font-size: 2rem;
}

.c16 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c16 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #dddddd;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c16 .language-plaintext {
  font-size: 1.6rem;
  background: #000000;
  color: #ffffff;
}

.c16 p,
.c16 pre {
  margin: 4.0rem 0;
}

.c16 pre {
  font-family: monospace;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
  background: #000000;
  color: #ffffff;
  padding: 2.4rem;
}

.c16 a,
.c16 a:visited,
.c16 a:link {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c16 a:hover {
  -webkit-filter: brightness(60%);
  filter: brightness(60%);
}

.c16 img {
  max-width: 100%;
}

.c16 .image {
  background: #dddddd;
  line-height: 0;
  margin: 4.0rem 0;
}

.c16 .image figcaption {
  padding: .8rem;
  text-align: center;
  font-size: 1.6rem;
  line-height: 1.3;
}

.c16 .image-style-side {
  float: right;
  max-width: 50%;
  margin: 2.4rem;
}

.c16 ul,
.c16 ol {
  margin: 4.0rem;
}

.c16 hr {
  border: none;
  border-bottom: 0.1rem solid #dddddd;
}

.c16 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c16 table {
  width: 100%;
  border-collapse: collapse;
}

.c16 table td,
.c16 table th {
  padding: 1.6rem;
  border: 0.1rem solid #dddddd;
}

.c10 {
  margin-top: 6.4rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c14 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width:768px) {
  .c0 {
    left: -0.4rem;
  }
}

@media (max-width:768px) {
  .c16 {
    font-size: 2rem;
  }

  .c16 .image-style-side {
    float: none;
    max-width: 100%;
    margin: 0;
  }
}

<div>
  <div
    class=""
  >
    <button
      aria-label="Open/Close"
      class="c0"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              href="/"
            >
              <a
                class="c7"
                target="_self"
              >
                <img
                  alt="Logo"
                  src="assets/images/I-am.jfif"
                />
              </a>
            </a>
          </h1>
        </div>
        <ul
          class="c8"
        >
          <li
            class=""
          >
            <a
              class="c9"
              href="#about"
              target="_self"
            >
              About
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="c9"
              href="#skills"
              target="_self"
            >
              Skills
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="c9"
              href="#contact"
              target="_self"
            >
              Contact
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="c10"
    >
      <header
        class="c11"
      >
        <h1
          class="c5 c6"
        >
          <a
            href="/"
          >
            <a
              class="c7"
              target="_self"
            >
              <img
                alt="Bruno Alexandre - Blog de ideias e de projetos realizados"
                src="assets/images/I-am.jfif"
              />
            </a>
          </a>
        </h1>
        <div
          class="c12"
        >
          <h2
            class="c5 c6"
          >
            Bruno Alexandre
          </h2>
          <p>
            Blog de ideias e de projetos realizados
          </p>
        </div>
      </header>
    </div>
    <div
      class="c13"
    >
      Ol√° Mundo
    </div>
    <div
      class="c14"
    >
      <footer
        class="c15"
      >
        <div
          class="c16"
        >
          feito com amor
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c17"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
